apm-agent-api/src/main/java/co/elastic/apm/api/ElasticApm.java

```diff
@@ -249,4 +249,44 @@ public static void captureException(@Nullable Throwable e) {
     public static void setServiceInfoForClassLoader(@Nullable ClassLoader classLoader, @Nullable String serviceName, @Nullable String serviceVersion) {
         // co.elastic.apm.api.ElasticApmInstrumentation.SetServiceInfoForClassLoader.setServiceInfoForClassLoader
     }
+
+    /**
+     * Returns the current value of a config option (regardless of however and whenever that option was (re)set)
+     * <p>
+     * This applies to dynamically updated values too, so for example "transaction_sample_rate"
+     * can be dynamically reset (eg by executing
+     *   System.setProperty("elastic.apm.transaction_sample_rate", "0.6")
+     * or changing the "transaction_sample_rate" in the property file or from fleet UI)
+     * and calling getConfig("transaction_sample_rate") would provide the current value
+     * </p>
+     * <p>
+     * Note the value returned can be null if an invalid key is used, and also
+     * if the "public-api" set of instrumentations has not been applied
+     * </p>
+     *
+     * @param key the string option
+     * @return The current value for the option, per the option type (String, Long, Integer, etc)
+     * @since 1.37.0
+     */
+    public static Object getConfig(String key) {
```

Comment:
```suggestion
    public static <T> T getConfig(String key, Class<T> type) {
```
[minor] this avoids casts in the calling code, and it allows to check the actual type at invocation time.

Reply:
excellent suggestion, thanks! changed now